test/typescript/dispatch.ts

```diff
@@ -1,12 +1,16 @@
 import {Dispatch, Action} from "../../index.d.ts";
 
 
-declare const dispatch: Dispatch;
-
+declare const dispatch: Dispatch<any>;
 
 const dispatchResult: Action = dispatch({type: 'TYPE'});
 
-
-type Thunk<O> = () => O;
+// thunk
+declare module "../../index.d.ts" {
+    export interface Dispatch<S> {
+        <R>(asyncAction: (dispatch: Dispatch<S>, getState: () => S) => R): R;
+    }
+}
```

Comment:
As far as I can see from https://github.com/Microsoft/TypeScript/pull/6213 external module augmentation should work by doing
```ts
// redux-thunk/index.d.ts
declare module "redux" {
  export interface Dispatch<S> {
    // ...
  }
}
```

@Igorbek could you please check this? If it works, I think this PR is good to merge.